lms/services/assignment.py

```diff
@@ -267,35 +263,19 @@ def get_assignments(  # noqa: PLR0913
         :param h_userids: return only assignments where these users are members.
         :param assignment_ids: return only assignments with these IDs.
         """
-        query = select(Assignment).where(Assignment.title.is_not(None))
+        candidate_courses = CourseService.courses_permission_check_query(
```

Comment:
Select all courses the current user has access to.

We pass the `course_id` filter here to avoid having to filter again by courses.